x-pack/legacy/plugins/alerting/server/alerts_client.ts

```diff
@@ -395,12 +390,32 @@ export class AlertsClient {
       );
       const scheduledTask = await this.scheduleAlert(id, attributes.alertTypeId);
       await this.savedObjectsClient.update('alert', id, { scheduledTaskId: scheduledTask.id });
-      await this.invalidateApiKey({ apiKey: attributes.apiKey });
```

Comment:
In some cases we still need the invalidate.  This assumes that prior to calling `enable()` no API key exists. 
I tested this with detection rules, and this results in dangling API keys because  sometimes an alert is updated before calling `enable()`:

1.  Create a custom rule, with the rule disabled.  This will call `create()` and generate an API key `k1`
2. Enable the rule
  1.  calls `update()`,  this creates  a new API key `k2` and invalidates the  old one `k1`
  2.  calls `enable()`, which in this case creates a new key `k3`.  `k2` doesn't get  invalidated.

Even after fixing #57036,  if `update()` is called before `enable()` you will end up with a dangling key.

Reply:
Thanks for the review 🙏 I see what you mean, I've also noticed this strange behaviour when fixing #57036 and created #57099 from that. This would solve the problem, but it never hurts to add a check here as well so I will do the change 👍

Reply:
Fixed in https://github.com/elastic/kibana/pull/56634/commits/f1367ec2270f5067f66c6b5ef631f4d4282eee2b.